R/dbWriteStream_DBIConnection_character_RecordBatchReader.R

```diff
@@ -0,0 +1,26 @@
+#' @rdname hidden_aliases
+#' @usage NULL
+dbWriteStream_DBIConnection_character_RecordBatchReader <- function(conn, name, value, append = FALSE, overwrite = FALSE, ...) {
+  require_arrow()
+
+  if (overwrite && append) {
+    stop("overwrite and append cannot both be TRUE")
+  }
+
+  if (overwrite || !append) {
+    dbWriteTable(conn, name, as.data.frame(value$read_next_batch()), ..., append = append, overwrite = overwrite)
+  }
+
+  while(TRUE) {
```

Comment:
This gave me pause. I expected an `else while`, but the code seems correct as is. Perhaps this and the preceding `if` could use a comment that explains the logic?